aten/src/ATen/native/cuda/BinaryMulDivKernel.cu

```diff
@@ -162,7 +163,7 @@ void div_floor_kernel_cuda(TensorIteratorBase& iter) {
             floordiv += scalar_t(1.0);
           }
         } else {
-          floordiv = std::copysign(scalar_t(0), a / b);
+          floordiv = ::c10::cuda::compat::copysignfp16(scalar_t(0), scalar_t(a / b));
```

Comment:
I might be dense, but is there a reason why calling copysignfp16 is the right thing here?

In the end, I would expect CUDA to use the GPU version so it won't matter, but this will be called with non-fp16 arguments and I think the naming is misleading and we're adding code that may be dead but is also likely not correct.

In my view it would be preferable to either make the CPU workaround version available in a header and punt to that from CUDAMathCompat.h for CPU or we should just raise a runtime error in the non `__CUDA_ARCH__` case to express that we don't want to run the code as a non-device function.

Reply:
Yes good catch, and also thanks for your suggestions. I think throwing an exception if called on CPU is better approach than trying to replicate the logic for host side. Will close this PR in favor of your updated one.